service/matching/tasklist/matcher.go

```diff
@@ -156,11 +154,6 @@ func (tm *TaskMatcher) Offer(ctx context.Context, task *InternalTask) (bool, err
 			}
 		}
 
-		if rsv != nil {
-			// there was a ratelimit token we consumed
-			// return it since we did not really do any work
-			rsv.Cancel()
```

Comment:
[canceling a token after its delay never does anything](https://cs.opensource.google/go/x/time/+/refs/tags/v0.5.0:rate/rate.go;l=176) (timeToAct is before "now"), and this was only ever returned after sleeping for the full delay, so this was pointless :|

Reply:
added a test to show that this did not serve a purpose in the original code